### PR TITLE
Add missing migrations for longtext and indices

### PR DESCRIPTION
**Pull request by PVince81**
- set type to "longtext" for "subjectparams" and "messageparams" by increasing length
- add index "activity_time" on column "timestamp"
- add index "activity_object" on columns "object_type" and "object_id"

Fixes https://github.com/owncloud/activity/issues/652

- ~~BUG: cannot change from text to longtext because Doctrine ignores length changes, possibly related unmerged upstream PR: https://github.com/doctrine/dbal/pull/3182, also see https://github.com/doctrine/dbal/issues/2566~~ => now resorting to direct SQL

### CODE REVIEW

**Comment by PVince81** [comment]
Core PR to implement the same code from upstream: https://github.com/owncloud/core/pull/33231.
This helps fix the issue but we can't automatically test this due to DBAL limitations.

**Comment by PVince81** [comment]
I guess it won't be possible to make Oracle comply without gymnastics similar to https://github.com/owncloud/activity/blob/master/appinfo/Migrations/Version20170131134507.php#L18

**Comment by DeepDiver1975** [comment]
> Core PR to implement the same code from upstream: [owncloud/core#33231](https://github.com/owncloud/core/pull/33231).
> This helps fix the issue but we can't automatically test this due to DBAL limitations.

for the oracle part we could leave a release note and tell people what to do ... there are only a few installations .....

**Comment by DeepDiver1975** [comment]
> Core PR to implement the same code from upstream: [owncloud/core#33231](https://github.com/owncloud/core/pull/33231).
> This helps fix the issue but we can't automatically test this due to DBAL limitations.

for the oracle part we could leave a release note and tell people what to do ... there are only a few installations .....

**Comment by PVince81** [comment]
@DeepDiver1975 in my local tests with core I noticed that Oracle would anyway convert the column to CLOB when asking for longtext. It looks like https://github.com/owncloud/activity/blob/master/appinfo/Migrations/Version20170131134507.php#L18 is already doing that. This means that we don't need to care about Oracle in activity for the longtext columns.

**Comment by PVince81** [comment]
had a chat with @DeepDiver1975 and due to the missing DBAL feature and also the fact that this kind of migration is seldom, we'll write a migration containing direct SQL statements instead, which removes the need for the core workaround PR.

**Comment by PVince81** [comment]
Seems PostgreSQL has no distinction between longtext and text, it's all "text" so we likely also do not need any change.

**Comment by PVince81** [comment]
updated with plain SQL statement for MySQL.

I'll have a test with PostgreSQL just to make sure there aren't other discrepancies and to confirm that no SQL change is needed.

**Comment by PVince81** [comment]
I've checked with PostgreSQL and apparently the column is already "text" and has no limit:
```
                                                                Table "public.oc_activity"
    Column     |          Type           | Collation | Nullable |                     Default                      | Storage  | Stats target | Des
cription 
---------------+-------------------------+-----------+----------+--------------------------------------------------+----------+--------------+----
---------
 activity_id   | bigint                  |           | not null | nextval('oc_activity_activity_id_seq'::regclass) | plain    |              | 
 timestamp     | integer                 |           | not null | 0                                                | plain    |              | 
 priority      | integer                 |           | not null | 0                                                | plain    |              | 
 type          | character varying(255)  |           |          | NULL::character varying                          | extended |              | 
 user          | character varying(64)   |           |          | NULL::character varying                          | extended |              | 
 affecteduser  | character varying(64)   |           | not null |                                                  | extended |              | 
 app           | character varying(255)  |           | not null |                                                  | extended |              | 
 subject       | character varying(255)  |           | not null |                                                  | extended |              | 
 subjectparams | text                    |           | not null |                                                  | extended |              | 
 message       | character varying(255)  |           |          | NULL::character varying                          | extended |              | 
 messageparams | text                    |           |          |                                                  | extended |              | 
 file          | character varying(4000) |           |          | NULL::character varying                          | extended |              | 
 link          | character varying(4000) |           |          | NULL::character varying                          | extended |              | 
 object_type   | character varying(255)  |           |          | NULL::character varying                          | extended |              | 
 object_id     | bigint                  |           | not null | 0                                                | plain    |              | 
```

so we only need to handle MySQL as already in the current PR code.

File: appinfo/Migrations/Version20181019151118.php [hunk 1]
```diff
@@ -0,0 +1,58 @@
+<?php
+/**
+ * @author Vincent Petry <pvince81@owncloud.com>
+ *
+ * @copyright Copyright (c) 2018, ownCloud, Inc.
```
Comment (DeepDiver1975) [inline appinfo/Migrations/Version20181019151118.php]: GmbH

File: appinfo/Migrations/Version20181022150134.php [hunk 2]
```diff
@@ -0,0 +1,58 @@
+<?php
+/**
+ * @author Vincent Petry <pvince81@owncloud.com>
+ *
+ * @copyright Copyright (c) 2018, ownCloud, Inc.
```
Comment (DeepDiver1975) [inline appinfo/Migrations/Version20181022150134.php]: GmbH

File: appinfo/Migrations/Version20181019151118.php (hunk 1 above)
Comment (PVince81) [reply appinfo/Migrations/Version20181019151118.php]: I copy-pasted this file from another migration, so it's wrong there as well.

I'll adjust them all then

**Comment by PVince81** [comment]
fixed license row

**Pull request merged**